### PR TITLE
Tuned CI/CD for FIPs build

### PR DESCRIPTION
**Pull request by zvonand**
This enables building via workflows of FIPS build (with BoringSSL)
Changes:
- Different Docker image for building (installed Docker-in-Docker)
- Additional flags for packager

### CODE REVIEW

File: docker/packager/binary/Dockerfile [hunk 1]
```diff
@@ -3,6 +3,53 @@
 ARG FROM_TAG=latest
 FROM clickhouse/test-util:$FROM_TAG
 
+# Rust toolchain and libraries
+ENV RUSTUP_HOME=/rust/rustup
```
Comment (Enmk) [inline docker/packager/binary/Dockerfile:7]: Why do we need rust toolchain?

File: docker/packager/packager [hunk 2]
```diff
@@ -90,7 +90,7 @@ def run_docker_image_with_env(
         user = f"{os.geteuid()}:{os.getegid()}"
 
     cmd = (
-        f"docker run --network=host --user={user} --rm --volume={output}:/output "
+        f"docker run -v /var/run/docker.sock:/var/run/docker.sock --network=host --user={user} --rm --volume={output}:/output "
```
Comment (Enmk) [inline docker/packager/packager:93]: We need to add a comment explaining that this is required for docker-in-docker required by the way we build BoringSSL libraries in FIPS mode.

File: docker/packager/binary/Dockerfile (hunk 1 above)
Comment (zvonand) [reply docker/packager/binary/Dockerfile:7]: Well, I took the latest Dockerfile from upstream. Maybe we can omit this, but let's first check it really works

Comment (zvonand) [reply docker/packager/binary/Dockerfile]: Removed

**Comment by vzakaznikov** [comment]
@zvonand, let's add a description to this PR :).

**Review by Enmk — COMMENTED** [review]
Needs some clarifications and also maybe cleanup?

File: docker/packager/packager [hunk 3]
```diff
@@ -190,6 +194,10 @@ def parse_env_variables(
         cc = compiler
         result.append("DEB_ARCH=amd64")
 
+    if fips:
+        cmake_flags.append("-DLINKER_NAME=/usr/lib/llvm-15/bin/ld.lld")
```
[the remark this answers is not in the document]
Comment (zvonand) [reply docker/packager/packager]: Isn't it installed alongside llvm?
I fixed the duplicated, now 14 is used

File: docker/packager/binary/build.sh [hunk 4]
```diff
@@ -1,4 +1,26 @@
 #!/usr/bin/env bash
+
+set -e
+
+# In case of test hung it is convenient to use pytest --pdb to debug it,
+# and on hung you can simply press Ctrl-C and it will spawn a python pdb,
+# but on SIGINT dockerd will exit, so ignore it to preserve the daemon.
+trap '' INT
+dockerd --host=unix:///var/run/docker.sock --host=tcp://0.0.0.0:2375 --default-address-pool base=172.17.0.0/12,size=24 &
```
Comment (Enmk) [inline docker/packager/binary/build.sh:9]: Where do these values (`172.17.0.0` and `:2375`) come from?

Comment (zvonand) [reply docker/packager/binary/build.sh:9]: 2375 is a default port for docker. 
this IP is semi-default IP pool for composing networks

**Pull request merged**